python/lsst/pipe/base/pipeline.py

```diff
@@ -111,6 +112,18 @@ def __str__(self):
         rep += ")"
         return rep
 
+    def __eq__(self, other: TaskDef) -> bool:
+        return self.taskName == other.taskName and\
+            self.config == other.config and\
+            self.taskClass == other.taskClass and\
+            self.label == other.label
+
+    def __hash__(self):
+        with io.StringIO() as sio:
+            self.config.saveToStream(sio)
+            configHash = hash(sio.getvalue())
+        return hash(self.taskName) ^ configHash ^ hash(self.taskClass) ^ hash(self.label)
```

Comment:
Hmm, usually people do `hash(tuple(....))` on multiple values, I don't know if `^` is better or worse.

Reply:
So it is close to the same thing, but without the overhead of creating temporary containers. This is what is recommended in official documentation for `__hash__` in python (though its not exclusively restricted to this method)

Reply:
It seems like the suggestion has changed sometime between python 3.4 (the last time I looked at it) and python 3.8 docs. I will update to use the new suggestion of making a tuple